screenshots/Screenshots.ts

```diff
@@ -1,48 +1,128 @@
+import childProcess from 'child_process';
+import fs from 'fs';
 import os from 'os';
+import path from 'path';
 
 import { expect } from '@playwright/test';
+import dayjs from 'dayjs';
 
 import { NavPage } from '../e2e/pages/nav-page';
 import { PreferencesPage } from '../e2e/pages/preferences';
 
-import type { Page, PageScreenshotOptions } from '@playwright/test';
+import type { Page } from '@playwright/test';
 
 interface ScreenshotsOptions {
   directory?: string;
+  isOsCommand?: boolean;
 }
 
 export class Screenshots {
+  private isOsCommand = true;
+  private sleepDuration = Number(process.env.RD_ENV_SCREENSHOT_SLEEP) || 1000;
+
+  // used by Mac api
+  private appBundleTitle = 'Electron';
+
+  protected windowTitle = '';
   private screenshotIndex = 0;
   readonly page: Page;
   readonly directory: string | undefined;
+  readonly screenshotDirectory: string | undefined;
 
   constructor(page: Page, opt?: ScreenshotsOptions) {
     this.page = page;
     if (opt?.directory) {
       const { directory } = opt;
 
       this.directory = directory === undefined || directory === null ? '' : `${ directory }/`;
+      this.screenshotDirectory = path.resolve(__dirname, 'output', os.platform(), this.directory);
+    }
+    if (opt?.isOsCommand) {
```

Comment:
If `opt?.isOsCommand` is `false` here, `this.isOsCommand` will end up being set to `true` because that is the default. I don't think that is what we want?

Reply:
I think we can just do the assignment when `typeof (opt?.isOsCommand) !== 'undefined'` to produce what is desired.